src/main/java/de/samply/common/ldmclient/centraxx/utils/PercentageLogger.java

```diff
@@ -0,0 +1,56 @@
+package de.samply.common.ldmclient.centraxx.utils;
+
+import org.slf4j.Logger;
+
+public class PercentageLogger {
+
+  private Logger logger;
+  private int numberOfElements;
+  private int counter = 0;
+  private int lastPercentage = 0;
+
+  /**
+   * Calculates percentage of remaining steps and logs it.
+   *
+   * @param logger logger
+   * @param numberOfElements number of elements
+   * @param description description of the operation
+   */
+  public PercentageLogger(Logger logger, int numberOfElements, String description) {
+
+    this.logger = logger;
+    this.numberOfElements = numberOfElements;
+    if (numberOfElements > 0) {
```

Comment:
would assert that the number of elements is positive in the constructor

Reply:
How would you assert it? Throwing an Illegal Argument Exception?

Reply:
Ja genau beispielsweise mit [Guava Preconditions](https://guava.dev/releases/21.0/api/docs/com/google/common/base/Preconditions.html).